go/private/rules/test.bzl

```diff
@@ -86,13 +85,15 @@ def _go_test_impl(ctx):
     external_srcs = split_srcs(external_source.srcs).go
 
     # now generate the main function
-    if ctx.attr.rundir:
-        if ctx.attr.rundir.startswith("/"):
-            run_dir = ctx.attr.rundir
-        else:
-            run_dir = pkg_dir(ctx.label.workspace_root, ctx.attr.rundir)
+    repo_relative_rundir = ctx.attr.rundir
+    if not repo_relative_rundir:
+        repo_relative_rundir = ctx.label.package if ctx.label.package else "."
+    if ctx.label.workspace_name:
+        # The test is contained in an external repository. The test runner cd's into the directory
```

Comment:
If the main workspace has a name, this would walk up and down to the same directory, which is fine, but the comment is misleading.

Reply:
Even if the main workspace has a name, the value in this line should always be empty. Added a comment explaining that.